.github/workflows/check_license_header.yaml

```diff
@@ -0,0 +1,79 @@
+                   -o  -name '*.h.in'                                             \
+                   -o  -name '*.tpp'                                              \
+                   -o  -name '*.cu'                                               \
+                   -o  -name 'CMakeLists.txt'                                     \
+               ')'                                                                \
+            > result-extension.check
+          # Genearate an error message for each file with an unkown extension
+          for filepath in `cat result-extension.check`; do                        \
+            echo "::error file=$filepath,line=1::check extension of $filepath";   \
+          done
+
+          test ! -s result-extension.check
+
+      - name: Check source code
+        if: always()          # needed so that it runs even if previous one fails
+        run: |
+          # Compare first lines of each source file with reference in misc/HEADER
+          find src include test miniapp -type f                                   \
+               '('                                                                \
+                       -name '*.cpp'                                              \
+                   -o  -name '*.h'                                                \
+                   -o  -name '*.h.in'                                             \
+                   -o  -name '*.tpp'                                              \
+                   -o  -name '*.cu'                                               \
+               ')'                                                                \
+            | xargs -I{} sh -c                                                    \
+                "head -n9 {} | diff -sq - misc/HEADER > /dev/null || echo {}"     \
+            > result-cpp.check
+
+          # Genearate an error message for each offending file
+          for filepath in `cat result-cpp.check`; do                              \
+            echo "::error file=$filepath,line=1::check the license in $filepath"; \
+          done
+
+          test ! -s result-cpp.check
+
+      - name: Check CMake files
+        if: always()          # needed so that it runs even if previous one fails
+        run: |
+          # Generate header license CMake compliant
+          sed 's|^//|#|g' misc/HEADER > HEADER_CMAKE
+
+          # Compare first lines of cmake files with the reference
+          find  '(' -type f ! -path './misc*' ')'                                 \
+                '('                                                               \
+                        -name 'CMakeLists.txt'                                    \
+                    -o  -name 'DLAF*.cmake'                                       \
+                    -o  -name 'DLAF*.cmake.in'                                    \
+                ')'                                                               \
```

Comment:
Good point. In case, the check for them is slightly different because they may (or may not ) have the shebang at first line. I'm going to add this check somehow.

About the extension check, I'm not really sure I got the point. Currently what the ext check does is to look inside `src`/`include`/`test` folders for files that have an extension different from ones listed there. It is a bit the "other way around", because it's not checking that C++ files have valid extension, but that inside there there are just files with valid extensions.

Anyway, we could decide to change it to rely on `file` output (iirc it does not rely on the extension), so that any file that is considered a C/C++ file, should have the extensions we want.

And at that point, we could also add the check for python files having `*.py`. However I'm not sure if this is what you have in mind.

Reply:
> About the extension check, I'm not really sure I got the point. Currently what the ext check does is to look inside `src`/`include`/`test` folders for files that have an extension different from ones listed there. It is a bit the "other way around", because it's not checking that C++ files have valid extension, but that inside there there are just files with valid extensions.
> 
> Anyway, we could decide to change it to rely on `file` output (iirc it does not rely on the extension), so that any file that is considered a C/C++ file, should have the extensions we want.
> 
> And at that point, we could also add the check for python files having `*.py`. However I'm not sure if this is what you have in mind.

Yeah, I didn't think clearly with the comment about the extensions. Somehow I was first thinking that you were looking to have a global whitelist-type of thing for allowed extensions in the repo, but it makes a lot less sense for python where the only reasonable extension is `.py`. For C++ it makes more sense to check the extensions since there are a lot of options, and I guess this was your main motivation for adding that check. So I think you can ignore my comment about the extensions...

Reply:
> For C++ it makes more sense to check the extensions since there are a lot of options, and I guess this was your main motivation for adding that check

Exactly! As I said it is a bit the other way around, because it is not

`C++ file => check extension is in the list`

but it is

`a file in src/include/test must be either a C++ or CMake file => check extension is in the list`

The alternative is to rely on `file` output, with something like

`if file tool says it is a c/c++ file => check extension is in the list`

Reply:
In my opinion what you have right now is reasonable and quite clear.